src/Jobs/PtpJob.php

```diff
@@ -0,0 +1,324 @@
+     * @param string $id Uuid associated to the job
+     *
+     */
+    public function __construct(
+        public int $volumeId,
+        public string $volumeName,
+        public string $inputFile,
+        public string $outputFile,
+        public User $user,
+        public string $id,
+    )
+    {
+        if (config('ptp.ptp_fail_construct') && config('ptp.ptp_storage_disk') == 'test') {
+            throw new Exception("Error!");
+        }
+
+        $this->volumeId = $volumeId;
+        $this->volumeName = $volumeName;
+        $this->outputFile = config('ptp.temp_dir').'/'.$outputFile;
+        $this->tmpInputFile = config('ptp.temp_dir').'/'.$inputFile.'.json';
+        $this->tmpImageInputFile = config('ptp.temp_dir').'/'.$inputFile.'_images.json';
+        $this->user = $user;
+        $this->id = $id;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $callback = function ($images, $paths){
+            $this->generateImageInputFile($paths,  $images);
+        };
+        $imageData = $this->generateInputFile();
+        $imageIds = array_keys($imageData);
+        $images = Image::whereIn('id', $imageIds)->get()->all();
+        FileCache::batch($images, $callback);
+        $this->python();
+        $this->uploadConvertedAnnotations();
+        $this->cleanupJob();
+        $this->user->notify(new PtpJobConcluded($this->volumeName));
+    }
+
+    /**
+     * Generate the input Job File
+     *
+     * @return
+     */
+    public function generateInputFile(): array
+    {
+        $imageAnnotationArray = [];
+
+        $pointShapeId = Shape::pointId(); //Find annotations with selected label in desired volume
+        $annotations = ImageAnnotation::join('image_annotation_labels','image_annotations.id', '=', 'image_annotation_labels.annotation_id')
+            ->join('images','image_annotations.image_id', '=','images.id')
+            ->where('images.volume_id', $this->volumeId)
+            ->where('image_annotations.shape_id', $pointShapeId)
+            ->select('image_annotations.id as id', 'images.id as image_id', 'image_annotations.points as points','image_annotations.shape_id as shape_id', 'image_annotation_labels.label_id as label_id')
+            ->get();
+
+
+        foreach ($annotations as $annotation) {
+            if (!isset($imageAnnotationArray[$annotation->image_id])) {
+                $imageAnnotationArray[$annotation->image_id] = [];
+            }
+            $imageAnnotationArray[$annotation->image_id][] = [
+                'annotation_id' => $annotation->id,
+                'points' => $annotation->points,
+                'shape' => $annotation->shape_id,
+                'image' => $annotation->image_id,
+                'label' => $annotation->label_id,
+            ];
+        };
+
+        //$inputFile.'.json' will be used for image annotations, $inputFile.'_images.json' for image paths
```

Comment:
I feel this comment belongs elsewhere because this method has nothing to do with the "images" input file.